src/celengine/astrocat.h

```diff
@@ -0,0 +1,69 @@
+
+#pragma once
+
+#include <climits>
+#include <string>
+#include <celutil/util.h>
+#include <fmt/printf.h>
+
+class AstroCatalog
+{
+ public:
+    typedef unsigned int IndexNumber;
+    static const IndexNumber InvalidIndex = UINT_MAX;
+    virtual IndexNumber getIndexNumberByName(const std::string&) = 0;
+    virtual std::string getNameByIndexNumber(IndexNumber) = 0;
+    virtual const std::string& getName() const = 0;
+};
+
+class SimpleAstroCatalog : public AstroCatalog
+{
+ protected:
+    std::string m_prefix;
+ public:
+    SimpleAstroCatalog(const std::string& pref) { m_prefix = pref; }
+    virtual IndexNumber getIndexNumberByName(const std::string&);
+    virtual std::string getNameByIndexNumber(IndexNumber);
+    virtual const std::string& getName() const;
+    const std::string& getPrefix() const { return m_prefix; };
+};
+
+class TychoAstroCatalog : public SimpleAstroCatalog
+{
+ public:
+    TychoAstroCatalog() : SimpleAstroCatalog("TYC") {}
+    virtual IndexNumber getIndexNumberByName(const std::string&);
+    virtual std::string getNameByIndexNumber(IndexNumber);
+};
+
+class CelestiaAstroCatalog : public SimpleAstroCatalog
```

Comment:
What the purpose of this class?

Reply:
Hm, I added it by convention, but in fact it's indeed unnecessary.